### PR TITLE
[WIP] Worker registration tests

### PR DESCRIPTION
**Pull request by aaannz**
After various tries of mocking and hacking various Mojo parts I found out I can use Test::Mojo and it's Mojo::UserAgent::Server to simulate WebUI and use `before_dispatch` hook to easily control desired response.

This is still WIP as api_call tests are not yet adapted to this framework, only registration tests are in. Even they are not complete since I do not test websocket setup failure yet.

### CODE REVIEW

File: lib/OpenQA/Worker/Common.pm [hunk 1]
```diff
@@ -29,13 +29,13 @@ our @EXPORT = qw($job $verbose $instance $worker_settings $pooldir $nocleanup
...
   STATUS_UPDATES_SLOW STATUS_UPDATES_FAST
-  add_timer remove_timer change_timer get_timer
+  add_timer remove_timer change_timer get_timer get_timers
   api_call verify_workerid register_worker ws_call);
 
 # Exported variables
 our $job;
 our $verbose  = 0;
-our $instance = 'manual';
+our $instance = 99;
```
Comment (okurz) [inline lib/OpenQA/Worker/Common.pm]: what does that mean (now)?

File: lib/OpenQA/Worker/Common.pm [hunk 2]
```diff
@@ -438,37 +443,47 @@ sub register_worker {
...
+                my $err_code = $tx->error->{code};
+                if ($err_code) {
+                    if ($err_code =~ /^4\d\d$/) {
+                        # don't retry when 4xx codes are returned. There is problem with scheduler
+                        log_error(
+                            sprintf(
+                                'ignoring server - server refused with code %s: %s',
+                                $tx->error->{code},
+                                $tx->res->body
+                            ));
```
Comment (okurz) [inline lib/OpenQA/Worker/Common.pm]: maybe time for extracting a method for the sake of readability? ;-)

File: lib/OpenQA/Worker/Common.pm (hunk 1 above)
Comment (aaannz) [inline lib/OpenQA/Worker/Common.pm]: This means that now it is functional (hopefully) when `--instance` option is not passed to the worker script. It is by no means a limit, it's just different enough (IMO) for systems where you expect to run user started workers.

**Comment by coolo** [comment]
looks good to me, but I won't merge WIP - learned that lesson :)

**Comment by coolo** [comment]
as it's very likely it's no longer WIP - how about you rebase and remove the tag?

**Comment by coolo** [comment]
the workers register quite differently meanwhile, so if this wasn't rebased in half a year, I  don't think it's worth waiting yet another half year

**Pull request closed without merging**